python/instrumentation/openinference-instrumentation-llama-index/tests/openinference/instrumentation/llama_index/test_callback.py

```diff
@@ -0,0 +1,354 @@
+import asyncio
+import logging
+import random
+from contextlib import suppress
+from itertools import count
+from typing import Any, AsyncIterator, Dict, Generator, Iterable, Iterator, List, Tuple
+
+import openai
+import pytest
+from httpx import AsyncByteStream, Response, SyncByteStream
+from llama_index import ListIndex, ServiceContext
+from llama_index.callbacks import CallbackManager
+from llama_index.llms import OpenAI
+from llama_index.schema import Document, TextNode
+from openinference.instrumentation.llama_index import LlamaIndexInstrumentor
+from openinference.instrumentation.openai import OpenAIInstrumentor
+from openinference.semconv.trace import (
+    DocumentAttributes,
+    EmbeddingAttributes,
+    MessageAttributes,
+    OpenInferenceSpanKindValues,
+    SpanAttributes,
+    ToolCallAttributes,
+)
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk import trace as trace_sdk
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+from respx import MockRouter
+
+for name, logger in logging.root.manager.loggerDict.items():
+    if name.startswith("openinference.") and isinstance(logger, logging.Logger):
+        logger.setLevel(logging.DEBUG)
+        logger.handlers.clear()
+        logger.addHandler(logging.StreamHandler())
+
+
+@pytest.mark.parametrize(
+    "is_stream,is_async",
+    [
+        (False, False),
+        (False, True),
+        (True, False),
```

Comment:
Suggestion: Use `pytest.param` to give IDs to the sub-tests.

Reply:
discussed offline. it's not needed at the moment